Изменения в ДЗ

Artemius555 opened the pull request:
Почему-то не показывает мою работу, когда нажимаю в шторме "Run tech.html". 

Исправил ошибки, но не уверен, что правильно воспринял информации из прошлого реквеста.

Artemius555 committed:
Изменения в ДЗ

ivan-filchakov commented on tech/tech.html:
```diff
@@ -13,6 +13,18 @@
     <div class="tech__content">
       <div class="techBlock"><div class="button selectable">my button</div></div>
       <div class="techBlock"><div class="button button--full selectable">my button</div></div>
+        <div class="titled-card">
```
чтобы блок заработал в `tech.html` его нужно обернуть в div с классом `techBlock` и добавить класс `selectable`
Типа так:
```
      <div class="techBlock">
        <div class="titled-card selectable">
          .....
        </div>
      </div>
```

ivan-filchakov commented on tech/tech.html:
```diff
@@ -13,6 +13,18 @@
+        <div class="titled-card">
+          <a href="#">
+            <div class="titled-card__image">
+              <img src="https://images.prom.ua/3342374130_w640_h640_samokleyuschayasya-plitka-na.jpg">
```
В `<img>` не хватает атрибута `alt=""` - мелочь, но из-за этого страница может не пройти валидацию.

ivan-filchakov commented on tech/tech.html:
```diff
@@ -13,6 +13,18 @@
       <div class="techBlock"><div class="button button--full selectable">my button</div></div>
+        <div class="titled-card">
+          <a href="#">
+            <div class="titled-card__image">
```
`<а>` - строчный элемент, и в него нежелательно вкладывать элементы блочного типа типа других divов и т.д., + сейчас он еще и ломает вложенность по БЭМу. Более корректно будет положить его на один уровень с остальными элементами, обозвать, допустим, `titled-card__link` и растянуть на всю ширину карточки (пустой а с классом типа: `<a href="#" class="titled-card__link"></a>`).

ivan-filchakov commented on style.css:
```diff
@@ -11,3 +11,35 @@
+  z-index: 2;
+}
+
+a {
```
не стоит стилизовать ссылку через глобальный тег `a {...}`, т.к. это повлияет на все ссылки на сайте, а нам нужно работать только в пределах блока из задачи. 
чтоб правильно стилизовать и натянуть ссылку на карточку нам нужен селектор типа `.titled-card__link {...}`

ivan-filchakov commented on style.css:
```diff
@@ -11,3 +11,35 @@
 }
+
+.titled-card {
+  height: 100%;
```
у `.titled-card` полностью убрать высоту

ivan-filchakov commented on style.css:
```diff
@@ -11,3 +11,35 @@
+  position: relative;
+  border-radius: 35px;
+}
+
```
для картинки через селектор `.titled-card__image img {...}` задать ширину 100%. 
Тогда она будет корректно растягиваться на весь блок и не будет плющиться.

ivan-filchakov commented on style.css:
```diff
@@ -11,3 +11,35 @@
+  border-radius: 35px;
+  z-index: 1;
+}
+
```
Один элемент проще выравнивать чем два, поэтому предлагаю еще раз всё упростить и полностью удалить `.titled-card__blur` , а его блюр-эффекты и скругление отдать в `.titled-card__text`.

ivan-filchakov commented on style.css:
```diff
@@ -11,3 +11,35 @@
+
+.titled-card__blur {
+  position: absolute;
+  top: 76.19%;
```
`top: 76.19%` не нужен, для позиционирования достаточно одного `bottom: 0`. Чтобы задать гибкую высоту используй `min-height` или `padding`

ivan-filchakov reviewed (changes requested):
### все в кучу по изменениям:

чтобы блок заработал в `tech.html` его нужно обернуть в div с классом `techBlock` и добавить класс `selectable`
Типа так:
```
      <div class="techBlock">
        <div class="titled-card selectable">
          .....
        </div>
      </div>
```

В `<img>` не хватает атрибута `alt=""` - мелочь, но из-за этого страница может не пройти валидацию.

По поводу линка:
`<а>` - строчный элемент, и в него нежелательно вкладывать элементы блочного типа типа других divов и т.д., + сейчас он еще и ломает вложенность по БЭМу. Более корректно будет положить его на один уровень с остальными элементами, обозвать, допустим, `titled-card__link` и растянуть на всю ширину карточки (пустой а с классом типа: `<a href="#" class="titled-card__link"></a>`).
И еще, не стоит стилизовать ссылку через глобальный тег `a {...}`, т.к. это повлияет на все ссылки на сайте, а нам нужно работать только в пределах блока из задачи. Чтобы правильно стилизовать и натянуть ссылку на карточку нам нужен селектор типа `.titled-card__link {...}`

Как подключишь tech.html - увидишь что картинка и плашка немного развалились, я хз как это объяснить (может Виталий поможет разобрать), но чтоб все было ок, у `.titled-card` нужно полностью убрать высоту, а для картинки через селектор `.titled-card__image img {...}` задать ширину 100%. 
Тогда она будет корректно растягиваться на весь блок и не будет плющиться.

И последнее - плашка с подписью тоже разбежались. Один элемент проще выравнивать чем два, поэтому предлагаю еще раз всё упростить и полностью удалить `.titled-card__blur` , а его блюр-эффекты и скругление отдать в `.titled-card__text`. 
`top: 76.19%` не нужен, для позиционирования достаточно одного `bottom: 0`. Чтобы задать гибкую высоту используй `min-height` или `padding`

Artemius555 committed:
Внёс правки. Единственное не понял, как отцентровать текст по вертикали, сделал но хз правильно ли.

Artemius555 committed:
Внёс правки.

Единственное не понял, как отцентровать текст по вертикали, сделал но хз правильно ли.

Artemius555 committed:
Merge remote-tracking branch 'origin/titled-card' into titled-card

Artemius555 committed:
Финальные правки

ivan-filchakov reviewed (approved):
ok

ivan-filchakov reviewed (approved):
ok
